mbofui/app/js/controllers.js

```diff
@@ -296,6 +302,56 @@ ocellus.controller('mapController', ['$compile', '$scope', '$rootScope','$filter
     getEvents(url);
   };
 
+  $(document).on('click','.declareIntentPut', function(e){
+    $scope.intendPut($(this).attr('data-intention'), $(this).attr('data-event'),$(this).attr('data-respondent'), $(this).attr('data-intention-url')  );
+  });
+
+  $(document).on('click','.declareIntentPost', function(e){
+    $scope.intendPost($(this).attr('data-intention'), $(this).attr('data-event'));
+  });
+
+  $scope.intendPut = function (intention, targetEvent, respondent, intentionUrl) {
+    data = {
+      'intention': intention,
+      'event': targetEvent,
+      'respondent': respondent
+    };
+    Bof.IntendPut(intentionUrl, data).then(function(result) {
+      leafletData.getMap().then(function(map) {
+        map.closePopup();
+      });
+      Bof.GetIntentions('/api/intentions/?username=self').then(function(intentionsList) {
+        $scope.intentions=intentionsList;
+        if($scope.textOnly){
+          var intentionsAdded = intentionIncluded($scope.textEventsAll, intentionsList);
+          $scope.textEventsAll = intentionsAdded;
+          $scope.textEvents = intentionsAdded;
+        }
+      });
+    });
+  };
+  $scope.intendPost = function (intent, eventUrl) {
+    var eventId = _.last(_.compact(eventUrl.split('/')));
+    data = {
+        'intention': intent,
+        'event':eventUrl
+    };
+    var intentionsUrl = '/api/intentions/';
```

Comment:
I see the prefix is used here. Maybe we want to make this a constant as well. It's only used once though so not a necessity.